tools/nl/README.md

```diff
@@ -1 +1,75 @@
-Tools related to Natural Language query support in Data Commons.
+# Tools Related to Natural Language Query in Data Commons
+
+## Curate Stat Var Descriptions
+
+### Input File Structure
+
+Statistical variables and topics descriptions used for building sentence
+embeddings are stored in [embeddings/input](./embeddings/input/), where each sub
+folder represents one "embedding index". An embedding index represents a set of
+stat vars / topics and their descriptions that are used together for vector
+match of a query sentence. For example
+[embeddings/input/base](./embeddings/input/base/) contains the stat var
+description input csv files of all base Data Commons stat vars and topics.
+
+Each stat var description input CSV contains two columns:
+
+- dcid: The stat var/ topic DCID
+- setence: A set of descriptions of the dcid, concatenated by ";"
+
+### Curate Description
+
+In a nutshell, the stat var description should be "precise" and "consise". It
+should accurately articulate the meaning of the stat vars with no ambiguity, and
+should be consise without too much unwanted contexts.
+
+Keep in mind that the descriptions are meant to be compared with natural laguage
+queries. They should mimic and reflects real world intent.
+
+On the other hand, the description should reflect the stat var constraints (age,
+time range etc) so it can be distinguished from other similar stat vars.
+
+## Test and Validate Description
+
+### Embedding Eval Playground
+
+To add (update) a small amount of stat var descripitions and test against some
+sample queries, use the [embeddings eval
+tool](https://autopush.datacommons.org/nl/eval/embeddings).
+
+- Enter or upload the interested queries
+
+- Enter or upload the curated stat var description in the format as mentioned
+  above
+
+After clicking on "Apply" from each step, you can check the matched stat vars
+for the given queries. The result is grouped by existing indexes and the curated
+descriptions are applied on top of the existing descriptions.
```

Comment:
I think it is useful to clarify the following.... suppose we are adding a "SV1;description2" for an `SV1` that already exists in the index as "SV1;description1".  In this case, is "description2" replacing "description1" or is it in addition to "description1".

Reply:
Done